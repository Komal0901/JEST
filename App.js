import {Text, View} from 'react-native';
import React, {Component} from 'react';
import MyProp from '././src/components/MyProp';
export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      counter: 0,
    };
  }
  getData(x) {
    const sum = x * 10;
    this.setState({counter: this.state.counter + sum});
    return sum; // here we retun sum which contains the final output after function  call,so that we dont have to pass manual output in test file.
  }
  render() {
    return (
      <View>
        <Text>{this.state.counter}</Text>
        <MyProp name={'Komal'} age={22} />
      </View>
    );
  }
}

export default App;
