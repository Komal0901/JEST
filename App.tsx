import {Text, View} from 'react-native';
import React, {Component} from 'react';

export class App extends Component {
  getData(x: number) {
    return x * 10;
  }
  render() {
    return (
      <View>
        <Text>App</Text>
      </View>
    );
  }
}

export default App;
