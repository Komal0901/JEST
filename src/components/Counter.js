import React, {useState} from 'react';
import {Button, Text, View} from 'react-native';

const Counter = () => {
  const [counter, setCounter] = useState(0);
  return (
    <View>
      <Text testID={'counter'}>Counter: 0</Text>
      <Button
        title="+"
        testID={'increment'}
        onPress={() => setCounter(counter + 1)}
      />
    </View>
  );
};

export default Counter;
