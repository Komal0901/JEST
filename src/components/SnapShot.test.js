import 'react-native';
import React from 'react';
import App from '../../App';
import renderer from 'react-test-renderer';
import {render, fireEvent} from '@testing-library/react-native';

import Counter from './Counter';
test('Test render Snapshot', () => {
  const snapshot = renderer.create(<App />).toJSON();
  console.log(snapshot);
  expect(snapshot).toMatchSnapshot(); //match old code to new
});
// counter file test case

// describe('Counter', () => {
//   it('increments by 1 each time increment is pressed', () => {
//     const {getByTestId} = render(<Counter />);

//     const initialState = getByTestId('counter');
//     expect(initialState).not.toBeNull();

//     const increment = getByTestId('increment');
//     fireEvent.press(increment);

//     const oneState = getByTestId('counter');
//     expect(oneState).not.toBeNull();
//   });

//   // other tests...
// });
// test('function testing', () => {
//   const Appref = renderer.create(<App />).toJSON();
//   console.log('function:-------->>>>>', Appref);
//   const input = Appref.getData(2);
//   console.log(input);
//   //   expect(Appref.getData(2)).toEqual(20);
// });
