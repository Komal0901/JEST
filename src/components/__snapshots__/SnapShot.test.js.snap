// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render Snapshot 1`] = `
<View>
  <Text>
    0
  </Text>
  <View>
    <Text>
      Komal
    </Text>
    <Text>
      22
    </Text>
  </View>
</View>
`;
